app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User 
+  attr_reader :hash
+  
+  def initialize(session)
+    session[:selection_owner_hash] ||= User.generate_owner_hash
+    @hash = session[:selection_owner_hash]
+  end
+
+  def self.generate_owner_hash
+    SecureRandom.urlsafe_base64
+  end
+
+  def self.find_session(session)
+    User.new(session)
+  end
+end 
```

Comment:
Git prefers to have a trailing newline at the end of files. This is another thing Rubocop will usually take care of for you.